pkg/services/authz/zanzana/server/server_list.go

```diff
@@ -0,0 +1,143 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	authzextv1 "github.com/grafana/grafana/pkg/services/authz/zanzana/proto/v1"
+)
+
+func (s *Server) List(ctx context.Context, r *authzextv1.ListRequest) (*authzextv1.ListResponse, error) {
+	ctx, span := tracer.Start(ctx, "authzServer.List")
+	defer span.End()
+
+	if info, ok := typeInfo(r.GetGroup(), r.GetResource()); ok {
+		return s.listTyped(ctx, r, info)
+	}
+
+	return s.listGeneric(ctx, r)
+}
+func (s *Server) listTyped(ctx context.Context, r *authzextv1.ListRequest, info TypeInfo) (*authzextv1.ListResponse, error) {
+	// 1. check if subject has access through namespace because then they can read all of them
+	res, err := s.openfga.Check(ctx, &openfgav1.CheckRequest{
+		StoreId:              s.storeID,
+		AuthorizationModelId: s.modelID,
+		TupleKey: &openfgav1.CheckRequestTupleKey{
+			User:     r.GetSubject(),
+			Relation: mapping[utils.VerbGet],
+			Object:   newNamespaceResourceIdent(r.GetGroup(), r.GetResource()),
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if res.GetAllowed() {
+		return &authzextv1.ListResponse{All: true}, nil
+	}
+
+	// 2. List all resources user has access too
```

Comment:
Do we need a list of folders here as well?

Reply:
Not at the moment but it is easy to add when we do need it